java.lang.NoClassDefFoundError: Could not initialize class org.mockito.internal.util.MockUtil

I want to use testng + powermock + spring to run test code,my jar version is:
powermock  = 1.6.2
powermock-module-testng = 1.6.2 
mockito = 1.10.19
testng = 6.8.7
spring = 4.1.1.RELEASE
JDK = 1.8.0

And my test code as follows,and not extends AutoBaseTest, can run normal，AutoBaseTest contains
@ContextConfiguration(locations={"/test/resources/applicationContext-test.xml"})

`@PrepareForTest({test.class}) 
public class mockTest  extends AutoBaseTest {

```
@ObjectFactory
public IObjectFactory getObjectFactory() {

    return new PowerMockObjectFactory();
}


@Test
public final void test() throws Exception  {
    ......
}
```

}`

the Error stack ：
`FAILED CONFIGURATION: @BeforeClass springTestContextBeforeTestClass
java.lang.ExceptionInInitializerError
    at org.mockito.internal.exceptions.stacktrace.ConditionalStackTraceFilter.<init>(ConditionalStackTraceFilter.java:17)
    at org.mockito.exceptions.base.MockitoException.filterStackTrace(MockitoException.java:41)
    at org.mockito.exceptions.base.MockitoException.<init>(MockitoException.java:30)
    at org.mockito.exceptions.misusing.MockitoConfigurationException.<init>(MockitoConfigurationException.java:18)
    at org.mockito.internal.configuration.plugins.PluginLoader.loadImpl(PluginLoader.java:66)
    at org.mockito.internal.configuration.plugins.PluginLoader.loadPlugin(PluginLoader.java:24)
    at org.mockito.internal.configuration.plugins.PluginRegistry.<init>(PluginRegistry.java:12)
    at org.mockito.internal.configuration.plugins.Plugins.<clinit>(Plugins.java:11)
    at org.mockito.internal.util.MockUtil.<clinit>(MockUtil.java:24)
    at org.mockito.internal.configuration.injection.scanner.MockScanner.<init>(MockScanner.java:22)
    at org.mockito.internal.configuration.InjectingAnnotationEngine.injectMocks(InjectingAnnotationEngine.java:96)
    at org.powermock.api.mockito.internal.configuration.PowerMockitoInjectingAnnotationEngine.process(PowerMockitoInjectingAnnotationEngine.java:35)
    at org.powermock.api.extension.listener.AnnotationEnabler.injectSpiesAndInjectToSetters(AnnotationEnabler.java:60)
    at org.powermock.api.extension.listener.AnnotationEnabler.beforeTestMethod(AnnotationEnabler.java:55)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:498)
    at org.powermock.reflect.internal.WhiteboxImpl.performMethodInvocation(WhiteboxImpl.java:1873)
    at org.powermock.reflect.internal.WhiteboxImpl.invokeMethod(WhiteboxImpl.java:663)
    at org.powermock.reflect.Whitebox.invokeMethod(Whitebox.java:415)
    at org.powermock.modules.testng.internal.PowerMockTestNGMethodHandler.injectMocksUsingAnnotationEnabler(PowerMockTestNGMethodHandler.java:75)
    at org.powermock.modules.testng.internal.PowerMockTestNGMethodHandler.invoke(PowerMockTestNGMethodHandler.java:47)
    at test.java.com.sosoPay.servicetest.mockTest_$$_jvstb5f_0.springTestContextBeforeTestClass(mockTest_$$_jvstb5f_0.java)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:498)
    at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:84)
    at org.testng.internal.Invoker.invokeConfigurationMethod(Invoker.java:564)
    at org.testng.internal.Invoker.invokeConfigurations(Invoker.java:213)
    at org.testng.internal.Invoker.invokeConfigurations(Invoker.java:138)
    at org.testng.internal.TestMethodWorker.invokeBeforeClassMethods(TestMethodWorker.java:175)
    at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:107)
    at org.testng.TestRunner.privateRun(TestRunner.java:767)
    at org.testng.TestRunner.run(TestRunner.java:617)
    at org.testng.SuiteRunner.runTest(SuiteRunner.java:335)
    at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:330)
    at org.testng.SuiteRunner.privateRun(SuiteRunner.java:291)
    at org.testng.SuiteRunner.run(SuiteRunner.java:240)
    at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
    at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
    at org.testng.TestNG.runSuitesSequentially(TestNG.java:1224)
    at org.testng.TestNG.runSuitesLocally(TestNG.java:1149)
    at org.testng.TestNG.run(TestNG.java:1057)
    at org.testng.remote.RemoteTestNG.run(RemoteTestNG.java:111)
    at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:204)
    at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:175)
Caused by: java.lang.NullPointerException
    at org.mockito.internal.configuration.plugins.Plugins.getStackTraceCleanerProvider(Plugins.java:17)
    at org.mockito.internal.exceptions.stacktrace.StackTraceFilter.<clinit>(StackTraceFilter.java:21)
    ... 48 more

FAILED CONFIGURATION: @AfterMethod springTestContextAfterTestMethod(public final void test.java.com.sosoPay.servicetest.mockTest.test() throws java.lang.Exception)
java.lang.NoClassDefFoundError: Could not initialize class org.mockito.internal.util.MockUtil
    at org.mockito.internal.configuration.injection.scanner.MockScanner.<init>(MockScanner.java:22)
    at org.mockito.internal.configuration.InjectingAnnotationEngine.injectMocks(InjectingAnnotationEngine.java:96)
    at org.powermock.api.mockito.internal.configuration.PowerMockitoInjectingAnnotationEngine.process(PowerMockitoInjectingAnnotationEngine.java:35)
    at org.powermock.api.extension.listener.AnnotationEnabler.injectSpiesAndInjectToSetters(AnnotationEnabler.java:60)
    at org.powermock.api.extension.listener.AnnotationEnabler.beforeTestMethod(AnnotationEnabler.java:55)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:498)
    at org.powermock.reflect.internal.WhiteboxImpl.performMethodInvocation(WhiteboxImpl.java:1873)
    at org.powermock.reflect.internal.WhiteboxImpl.invokeMethod(WhiteboxImpl.java:663)
    at org.powermock.reflect.Whitebox.invokeMethod(Whitebox.java:415)
    at org.powermock.modules.testng.internal.PowerMockTestNGMethodHandler.injectMocksUsingAnnotationEnabler(PowerMockTestNGMethodHandler.java:75)
    at org.powermock.modules.testng.internal.PowerMockTestNGMethodHandler.invoke(PowerMockTestNGMethodHandler.java:47)
    at test.java.com.sosoPay.servicetest.mockTest_$$_jvstb5f_0.springTestContextAfterTestMethod(mockTest_$$_jvstb5f_0.java)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:498)
    at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:84)
    at org.testng.internal.Invoker.invokeConfigurationMethod(Invoker.java:564)
    at org.testng.internal.Invoker.invokeConfigurations(Invoker.java:213)
    at org.testng.internal.Invoker.invokeMethod(Invoker.java:786)
    at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:901)
    at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1231)
    at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:127)
    at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:111)
    at org.testng.TestRunner.privateRun(TestRunner.java:767)
    at org.testng.TestRunner.run(TestRunner.java:617)
    at org.testng.SuiteRunner.runTest(SuiteRunner.java:335)
    at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:330)
    at org.testng.SuiteRunner.privateRun(SuiteRunner.java:291)
    at org.testng.SuiteRunner.run(SuiteRunner.java:240)
    at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
    at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
    at org.testng.TestNG.runSuitesSequentially(TestNG.java:1224)
    at org.testng.TestNG.runSuitesLocally(TestNG.java:1149)
    at org.testng.TestNG.run(TestNG.java:1057)
    at org.testng.remote.RemoteTestNG.run(RemoteTestNG.java:111)
    at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:204)
    at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:175)

FAILED CONFIGURATION: @AfterClass springTestContextAfterTestClass
java.lang.NoClassDefFoundError: Could not initialize class org.mockito.internal.util.MockUtil
    at org.mockito.internal.configuration.injection.scanner.MockScanner.<init>(MockScanner.java:22)`

--- a/api/mockito/src/main/java/org/powermock/api/extension/listener/AnnotationEnabler.java
+++ b/api/mockito/src/main/java/org/powermock/api/extension/listener/AnnotationEnabler.java
@@ -51,9 +51,19 @@ public class AnnotationEnabler extends AbstractPowerMockTestListenerBase impleme
 
     @Override
     public void beforeTestMethod(Object testInstance, Method method, Object[] arguments) throws Exception {
+
+        // Mockito uses context class loader not only to load resources, but to load plugin implementers, but for
+        // other purpose default/application class loader is used. So PluginRepository and Plugin could be loaded by
+        // different class loaders. Fix will be
+
+        ClassLoader contextClassLoader = Thread.currentThread().getContextClassLoader();
+        Thread.currentThread().setContextClassLoader(this.getClass().getClassLoader());
+
         standardInject(testInstance);
         injectSpiesAndInjectToSetters(testInstance);
         injectCaptor(testInstance);
+
+        Thread.currentThread().setContextClassLoader(contextClassLoader);
     }
 
     private void injectSpiesAndInjectToSetters(Object testInstance) {

--- a/modules/module-impl/testng-agent/pom.xml
+++ b/modules/module-impl/testng-agent/pom.xml
@@ -20,7 +20,7 @@
         <dependency>
             <groupId>org.testng</groupId>
             <artifactId>testng</artifactId>
-            <version>6.8.21</version>
+            <version>6.9.10</version>
             <scope>compile</scope>
         </dependency>
         <dependency>

--- a/modules/module-impl/testng-agent/src/main/java/org/powermock/modules/testng/PowerMockObjectFactory.java
+++ b/modules/module-impl/testng-agent/src/main/java/org/powermock/modules/testng/PowerMockObjectFactory.java
@@ -29,14 +29,18 @@ import java.lang.reflect.Constructor;
  */
 public class PowerMockObjectFactory implements IObjectFactory {
 
-     static {
-        if(PowerMockObjectFactory.class.getClassLoader() != ClassLoader.getSystemClassLoader()) {
-            throw new IllegalStateException("PowerMockObjectFactory can only be used with the system classloader but was loaded by "+PowerMockObjectFactory.class.getClassLoader());
+    static {
+        if (PowerMockObjectFactory.class.getClassLoader() != ClassLoader.getSystemClassLoader()) {
+            throw new IllegalStateException("PowerMockObjectFactory can only be used with the system classloader but was loaded by " + PowerMockObjectFactory.class.getClassLoader());
         }
         PowerMockAgent.initializeIfPossible();
     }
 
-    private final ObjectFactoryImpl defaultObjectFactory = new ObjectFactoryImpl();
+    private final ObjectFactoryImpl defaultObjectFactory;
+
+    public PowerMockObjectFactory() {
+        defaultObjectFactory = new ObjectFactoryImpl();
+    }
 
     @Override
     public Object newInstance(Constructor constructor, Object... params) {

--- a/modules/module-impl/testng/src/main/java/org/powermock/modules/testng/internal/Assumes.java
+++ b/modules/module-impl/testng/src/main/java/org/powermock/modules/testng/internal/Assumes.java
@@ -1,0 +1,34 @@
+/*
+ *   Copyright 2016 the original author or authors.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+package org.powermock.modules.testng.internal;
+
+import org.testng.SkipException;
+
+/**
+ *
+ */
+public class Assumes {
+
+    public static void assumeTrue(String reason, boolean assertion) {
+        if (!assertion) {
+            throw new SkipException(reason);
+        }
+    }
+
+
+}

--- a/modules/module-test/mockito/testng/pom.xml
+++ b/modules/module-test/mockito/testng/pom.xml
@@ -43,6 +43,12 @@
                     <suiteXmlFiles>
                         <suiteXmlFile>suite.xml</suiteXmlFile>
                     </suiteXmlFiles>
+                    <systemProperties>
+                        <property>
+                            <name>mockitoVersion</name>
+                            <value>${mockito.version}</value>
+                        </property>
+                    </systemProperties>
                 </configuration>
             </plugin>
         </plugins>

--- a/modules/module-test/mockito/testng/src/test/java/samples/powermockito/testng/staticmocking/Mockito1MockStaticTest.java
+++ b/modules/module-test/mockito/testng/src/test/java/samples/powermockito/testng/staticmocking/Mockito1MockStaticTest.java
@@ -16,7 +16,10 @@
 package samples.powermockito.testng.staticmocking;
 
 import org.powermock.core.classloader.annotations.PrepareForTest;
-import org.powermock.modules.testng.PowerMockTestCase;
+import org.powermock.modules.testng.PowerMockObjectFactory;
+import org.testng.Assert;
+import org.testng.IObjectFactory;
+import org.testng.annotations.ObjectFactory;
 import org.testng.annotations.Test;
 import samples.singleton.StaticHelper;
 import samples.singleton.StaticService;
@@ -24,7 +27,6 @@ import samples.singleton.StaticService;
 import static org.powermock.api.mockito.PowerMockito.mockStatic;
 import static org.powermock.api.mockito.PowerMockito.verifyStatic;
 import static org.powermock.api.mockito.PowerMockito.when;
-import static org.testng.Assert.assertEquals;
 
 
 /**
@@ -32,10 +34,16 @@ import static org.testng.Assert.assertEquals;
  * static+final+native methods mocking.
  */
 @PrepareForTest({StaticService.class, StaticHelper.class})
-public class MockStaticTest extends PowerMockTestCase {
+public class Mockito1MockStaticTest {
+
+    @ObjectFactory
+    public IObjectFactory getObjectFactory() {
+        return new PowerMockObjectFactory();
+    }
 
     @Test
     public void testMockStatic() throws Exception {
+
         mockStatic(StaticService.class);
         String expected = "Hello altered World";
         when(StaticService.say("hello")).thenReturn("Hello altered World");
@@ -45,12 +53,13 @@ public class MockStaticTest extends PowerMockTestCase {
         verifyStatic();
         StaticService.say("hello");
 
-        assertEquals(expected, actual);
+        Assert.assertEquals(expected, actual);
     }
 
 
     @Test
     public void testMockStaticFinal() throws Exception {
+
         mockStatic(StaticService.class);
         String expected = "Hello altered World";
         when(StaticService.sayFinal("hello")).thenReturn("Hello altered World");
@@ -60,6 +69,6 @@ public class MockStaticTest extends PowerMockTestCase {
         verifyStatic();
         StaticService.sayFinal("hello");
 
-        assertEquals(expected, actual);
+        Assert.assertEquals(expected, actual);
     }
 }

--- a/modules/module-test/mockito/testng/src/test/java/samples/powermockito/testng/staticmocking/Mockito2MockStaticTest.java
+++ b/modules/module-test/mockito/testng/src/test/java/samples/powermockito/testng/staticmocking/Mockito2MockStaticTest.java
@@ -1,0 +1,79 @@
+/*
+ * Copyright 2008 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package samples.powermockito.testng.staticmocking;
+
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.testng.PowerMockObjectFactory;
+import org.testng.Assert;
+import org.testng.IObjectFactory;
+import org.testng.annotations.ObjectFactory;
+import org.testng.annotations.Test;
+import samples.singleton.StaticHelper;
+import samples.singleton.StaticService;
+
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.verifyStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+
+
+/**
+ * Test class to demonstrate static, static+final, static+native and
+ * static+final+native methods mocking.
+ */
+@PrepareForTest({StaticService.class, StaticHelper.class})
+public class Mockito2MockStaticTest {
+
+    @ObjectFactory
+    public IObjectFactory getObjectFactory() {
+        return new PowerMockObjectFactory();
+    }
+
+
+    @Test
+    public void testMockStatic() throws Exception {
+
+        System.out.println("Skip test while Mockito doesn't deliver fix");
+
+        mockStatic(StaticService.class);
+        String expected = "Hello altered World";
+        when(StaticService.say("hello")).thenReturn("Hello altered World");
+
+        String actual = StaticService.say("hello");
+
+        verifyStatic();
+        StaticService.say("hello");
+
+        Assert.assertEquals(expected, actual);
+    }
+
+
+    @Test
+    public void testMockStaticFinal() throws Exception {
+
+        System.out.println("Skip test while Mockito doesn't deliver fix");
+
+        mockStatic(StaticService.class);
+        String expected = "Hello altered World";
+        when(StaticService.sayFinal("hello")).thenReturn("Hello altered World");
+
+        String actual = StaticService.sayFinal("hello");
+
+        verifyStatic();
+        StaticService.sayFinal("hello");
+
+        Assert.assertEquals(expected, actual);
+    }
+}

--- a/modules/module-test/mockito/testng/suite.xml
+++ b/modules/module-test/mockito/testng/suite.xml
@@ -1,8 +1,39 @@
-<suite name="dgf" verbose="10" object-factory="org.powermock.modules.testng.PowerMockObjectFactory">
-    <test name="dgf">
+<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >
+<suite name="PowerMockito TestNG" verbose="10" object-factory="org.powermock.modules.testng.PowerMockObjectFactory">
+    <test name="General">
         <classes>
             <class name="samples.powermockito.testng.staticmocking.MockStaticNotPreparedTest"/>
             <class name="samples.powermockito.testng.staticmocking.MockStaticPreparedWithoutMockStaticTest"/>
+        </classes>
+    </test>
+    <test name="Mockito1">
+        <method-selectors>
+            <method-selector>
+                <script language="beanshell">
+                    <![CDATA[
+                        String str = System.getProperty("mockitoVersion");
+                        str.startsWith("1");
+                      ]]>
+                </script>
+            </method-selector>
+        </method-selectors>
+        <classes>
+            <class name="samples.powermockito.testng.staticmocking.Mockito1MockStaticTest"/>
+        </classes>
+    </test>
+    <test name="Mockito2">
+        <method-selectors>
+            <method-selector>
+                <script language="beanshell">
+                    <![CDATA[
+                        String str = System.getProperty("mockitoVersion");
+                        str.startsWith("2")&&str.contains(".46");
+                      ]]>
+                </script>
+            </method-selector>
+        </method-selectors>
+        <classes>
+            <class name="samples.powermockito.testng.staticmocking.Mockito2MockStaticTest"/>
         </classes>
     </test>
 </suite>

--- a/tests/utils/src/main/java/samples/powermockito/MockitoVersion.java
+++ b/tests/utils/src/main/java/samples/powermockito/MockitoVersion.java
@@ -1,0 +1,49 @@
+/*
+ *   Copyright 2016 the original author or authors.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+package samples.powermockito;
+
+public class MockitoVersion {
+
+    private static final SystemPropertiesMockitoVersion MOCKITO_VERSION = new SystemPropertiesMockitoVersion();
+
+
+    public static boolean isMockito1(){
+        return MOCKITO_VERSION.isMockito1();
+    }
+
+    public static boolean isMockito2(){
+        return MOCKITO_VERSION.isMockito2();
+    }
+
+
+    private static class SystemPropertiesMockitoVersion  {
+        private final String version;
+
+        private  SystemPropertiesMockitoVersion(){
+            version = System.getProperty("mockitoVersion");
+        }
+
+        private boolean isMockito1(){
+            return version.startsWith("1");
+        }
+
+        public boolean isMockito2() {
+            return version.startsWith("2");
+        }
+    }
+}

